### PR TITLE
Sonarqube

### PR DESCRIPTION
**Pull request by volbo**

### CODE REVIEW

File: gift-project/service/service-impl/src/main/java/com/epam/esm/core/impl/validator/UserValidator.java [hunk 1]
```diff
@@ -119,18 +118,12 @@ public void isValidRole(EntityDTO role) {
         }
     }
 
-    private void isValidCreationDate(LocalDateTime creationDate) {
-        if (creationDate != null) {
+    private void isValidDate(LocalDateTime date) {
```
Comment (NMikle) [inline gift-project/service/service-impl/src/main/java/com/epam/esm/core/impl/validator/UserValidator.java:121]: please, rename to checkDateIsNull

File: gift-project/service/service-impl/src/main/java/com/epam/esm/core/impl/validator/TagValidator.java [hunk 2]
```diff
@@ -17,7 +16,7 @@
 @Component
 public class TagValidator implements Validator<TagDTO> {
 
-    public void isValid(TagDTO tagDTO) throws ServiceException {
+    public void isValid(TagDTO tagDTO) {
```
Comment (NMikle) [inline gift-project/service/service-impl/src/main/java/com/epam/esm/core/impl/validator/TagValidator.java:19]: please, rename to checkValidity

File: gift-project/persistence/persistence-api/src/main/java/com/epam/esm/data/api/repository/CertificateRepository.java [hunk 3]
```diff
@@ -29,7 +29,7 @@
      * @param isSortedByName true if sorting by certificate name should be done otherwise false
      * @return list of certificates that match criteria
      */
-    List<Certificate> findByCriteria(Integer limit, Integer offset, List<String> tagNames, String textPart, Boolean isSortedByName);
+    List<Certificate> findByCriteria(Integer limit, Integer offset, List<String> tagNames, String textPart, boolean isSortedByName);
```
Comment (NMikle) [inline gift-project/persistence/persistence-api/src/main/java/com/epam/esm/data/api/repository/CertificateRepository.java:32]: please, incapsulate these params into some object (e.g. `SearchCriteria`)

File: gift-project/persistence/persistence-impl/src/main/java/com/epam/esm/data/impl/repository/JpaCertificateRepository.java [hunk 4]
```diff
@@ -33,7 +33,7 @@ public JpaCertificateRepository(EntityManager entityManager) {
     }
 
     @Override
-    public List<Certificate> findByCriteria(Integer limit, Integer offset, List<String> tagNames, String textPart, Boolean isSortedByName) {
+    public List<Certificate> findByCriteria(Integer limit, Integer offset, List<String> tagNames, String textPart, boolean isSortedByName) {
```
Comment (NMikle) [inline gift-project/persistence/persistence-impl/src/main/java/com/epam/esm/data/impl/repository/JpaCertificateRepository.java:36]: please, incapsulate these params into some object (e.g. SearchCriteria)

File: gift-project/service/service-impl/src/main/java/com/epam/esm/core/impl/service/CertificateServiceImpl.java [hunk 5]
```diff
@@ -56,6 +55,9 @@
     @Value("${default.certificate_service.limit}")
     private int defaultServiceLimit;
 
+    @Value("${certificate.text}")
```
[the remark this answers is not in the document]
Comment (volbo) [reply gift-project/service/service-impl/src/main/java/com/epam/esm/core/impl/service/CertificateServiceImpl.java]: fixed

File: gift-project/service/service-impl/src/main/java/com/epam/esm/core/impl/service/UserServiceImpl.java [hunk 6]
```diff
@@ -72,6 +72,12 @@
     @Value("${default.user_service.limit}")
     private int defaultServiceLimit;
 
+    @Value("${user.text}")
```
[the remark this answers is not in the document]
Comment (volbo) [reply gift-project/service/service-impl/src/main/java/com/epam/esm/core/impl/service/UserServiceImpl.java]: fixed

File: gift-project/service/service-impl/src/main/java/com/epam/esm/core/impl/service/UserServiceImpl.java [hunk 7]
```diff
@@ -72,6 +72,12 @@
...
+    private String userText;
+
+    @Value("${not_present_database.text}")
```
[the remark this answers is not in the document]
Comment (volbo) [reply gift-project/service/service-impl/src/main/java/com/epam/esm/core/impl/service/UserServiceImpl.java]: fixed

File: gift-project/service/service-impl/src/main/java/com/epam/esm/core/impl/validator/UserValidator.java (hunk 1 above)
Comment (volbo) [reply gift-project/service/service-impl/src/main/java/com/epam/esm/core/impl/validator/UserValidator.java]: fixed

File: gift-project/service/service-impl/src/main/java/com/epam/esm/core/impl/validator/CertificateValidator.java [hunk 8]
```diff
@@ -62,7 +61,7 @@ public void isValid(CertificateDTO certificateDTO) throws ServiceException {
         isValidTags(certificateDTO.getTags());
     }
 
-    public void isValidIgnoreAuditDates(CertificateDTO certificateDTO) throws ServiceException {
+    public void isValidIgnoreAuditDates(CertificateDTO certificateDTO) {
```
[the remark this answers is not in the document]
Comment (volbo) [reply gift-project/service/service-impl/src/main/java/com/epam/esm/core/impl/validator/CertificateValidator.java]: fixed

File: gift-project/service/service-impl/src/main/java/com/epam/esm/core/impl/validator/OrderValidator.java [hunk 9]
```diff
@@ -10,20 +10,19 @@
...
     @Override
     public void isValid(OrderDTO orderDTO) {
         isValidCost(orderDTO.getCost());
-        isValidCreationDate(orderDTO.getCreationDate());
-        isValidChangedDate(orderDTO.getChangedDate());
+        isValidDate(orderDTO.getCreationDate());
+        isValidDate(orderDTO.getChangedDate());
     }
 
     public void isValidUser(OrderDTO orderDTO) {
```
[the remark this answers is not in the document]
Comment (volbo) [reply gift-project/service/service-impl/src/main/java/com/epam/esm/core/impl/validator/OrderValidator.java]: fixed

File: gift-project/service/service-impl/src/main/java/com/epam/esm/core/impl/validator/TagValidator.java (hunk 2 above)
Comment (volbo) [reply gift-project/service/service-impl/src/main/java/com/epam/esm/core/impl/validator/TagValidator.java]: fixed

File: gift-project/persistence/persistence-api/src/main/java/com/epam/esm/data/api/repository/CertificateRepository.java (hunk 3 above)
Comment (volbo) [reply gift-project/persistence/persistence-api/src/main/java/com/epam/esm/data/api/repository/CertificateRepository.java]: fixed

File: gift-project/persistence/persistence-impl/src/main/java/com/epam/esm/data/impl/repository/JpaCertificateRepository.java (hunk 4 above)
Comment (volbo) [reply gift-project/persistence/persistence-impl/src/main/java/com/epam/esm/data/impl/repository/JpaCertificateRepository.java]: fixed

File: gift-project/service/service-impl/src/main/java/com/epam/esm/core/impl/validator/OrderValidator.java [hunk 10]
```diff
@@ -34,18 +33,12 @@ private void isValidCost(BigDecimal cost) {
         }
     }
 
-    private void isValidCreationDate(LocalDateTime creationDate) {
-        if (creationDate != null) {
+    private void isValidDate(LocalDateTime date) {
```
[the remark this answers is not in the document]
Comment (volbo) [reply gift-project/service/service-impl/src/main/java/com/epam/esm/core/impl/validator/OrderValidator.java]: fixed

**Comment by NMikle** [comment]
Looks good to me!

**Comment by volbo** [comment]
Comments were fixed.

**Pull request closed without merging**